repository_service_tuf/helpers/tuf.py

```diff
@@ -32,12 +32,25 @@
 repository_metadata: Dict[str, Metadata] = {}
 
 
+@dataclass
+class KeyProp:
+    # "key": Any (Any follows the KEY_SCHEMA from securesystemslib)
+    key: Optional[Dict[str, Any]] = None
+    error: Optional[str] = None
+
+
+class KeyInput(TypedDict):
+    filename: str
+    password: str
+    key: KeyProp
```

Comment:
This could be a dataclass, and we could use the `asdict` to (de)serialize it. As I mentioned before, it would be more consistent with the already existent dataclasses. Not sure that KeyProp made it more clear.

Reply:
> I'm not sure if we need this dataclass here.
> Not sure that KeyProp made it more clear.

Seeing the code for the first time it was pretty difficult to find out the types of the different class properties and their keys.
e.g., where this ```k.get("key")):``` ([ref](https://github.com/vmware/repository-service-tuf-cli/blob/main/repository_service_tuf/cli/admin/ceremony.py#L295)) or ```k.get("path")``` ([ref](https://github.com/vmware/repository-service-tuf-cli/blob/main/repository_service_tuf/cli/admin/ceremony.py#L297), which by the way the ```path``` key did not exist in the RolesKeysInput.keys.values) keys came from.
I think KeyProp and KeyInput types clear up the types and their structure.

Reply:
Ok, I got your point.
Let's just add an explicit name for `KeyProp`.
I also think making `KeyInput` a dataclass to keep the standard and we can use `asdict` if necessary.
what do you think?

Reply:
> Let's just add an explicit name for ```KeyProp```.

What do you think about ```KeySchema```?

> I also think making ```KeyInput``` a dataclass to keep the standard and we can use ```asdict``` if necessary.

sounds good! (I didn't know about the asdict method before, nice)

Reply:
`KeySchema` looks good.

IMO, the problem is not that we are replacing typing as an object dataclass.

We make it a dataclass and need to adjust the data structure used in some other places to be consistent.

In `RolesKeyInput`, the `keys` is a required field, and when we initiate the `RolesKeyInput` without giving the keys, we use the default factory as a dict (before, we used a standard python Dictionary, and in your suggestion, we replace by a TypeDict that is compatible)

In practice, we do it in 
https://github.com/vmware/repository-service-tuf-cli/blob/5e9097862f2b98afd5d78196a8d8641213152485/repository_service_tuf/cli/admin/ceremony.py#L287-L290

An option here is make the `keys` field `keys: Optional[Dict[str, KeyInput]] = None`

Neat: I think even the `RolesKeysInput` name is not good anymore; probably, I started as 'key role schema', and it becomes more of a `RoleSettingsInput`.

Also, we need to adjust the usage when we parse the keys.

In the end, our options are
1) We move to dataclass and adjust everything
2) We could keep the dictionary
3) Use TypedDict, but still my last option, as it sounds more an workaround for option 1

Reply:
Thanks for the elaborate response @kairoaraujo, I see your point on how the classes evolved.
I think it makes sense to try to move to dataclass and adjust everything and it seems like the right time to do it.
I will update you with a new commit push 😉